src/LineaStateBridge.sol

```diff
@@ -1,17 +1,251 @@
+
+    /// @notice The address of the LinearWorldID contract on any Linea Stack chain
+    address public immutable lineaWorldIDAddress;
+
+    /// @notice address for Linea Stack chain Ethereum mainnet L1CrossDomainMessenger contract
+    address internal immutable crossDomainMessengerAddress;
+
+    /// @notice Ethereum mainnet worldID Address
+    address public immutable worldIDAddress;
+
+    /// @notice Amount of gas purchased on the Linea Stack chain for propagateRoot
+    uint32 internal _gasLimitPropagateRoot;
+
+    /// @notice Amount of gas purchased on the OLinea Stack chain for SetRootHistoryExpiry
+    uint32 internal _gasLimitSetRootHistoryExpiry;
+
+    /// @notice Amount of gas purchased on the Linea Stack chain for transferOwnershipOp
+    uint32 internal _gasLimitTransferOwnership;
+
+    /// @notice The default gas limit amount to buy on an Linea stack chain to do simple transactions
+    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 1_000_000;
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            EVENTS                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Emitted when the StateBridge gives ownership of the LineaWorldID contract
+    /// to the WorldID Identity Manager contract away
+    /// @param previousOwner The previous owner of the LineaWorldID contract
+    /// @param messageService The new owner of the LineaWorldID contract
+    /// @param remoteAddress Whether the ownership transfer is local (Linea/Linea Stack chain EOA/contract)
+    /// or an Ethereum EOA or contract
+    event OwnershipTransferredLinea(
+        address indexed previousOwner, address indexed messageService, address remoteAddress
+    );
+
+    /// @notice Emitted when the StateBridge sends a root to the LineaWorldID contract
+    /// @param root The root sent to the LineaWorldID contract on the Linea Stack chain
+    event RootPropagated(uint256 root);
+
+    /// @notice Emitted when the StateBridge sets the root history expiry for OpWorldID, LineaWorldID and PolygonWorldID
+    /// @param rootHistoryExpiry The new root history expiry
+    event SetRootHistoryExpiry(uint256 rootHistoryExpiry);
+
+    /// @notice Emitted when the StateBridge sets the gas limit for sendRootOp
+    /// @param _lineaGasLimit The new _lineaGasLimit  for sendRootOp
+    event SetGasLimitPropagateRoot(uint32 _lineaGasLimit);
+
+    /// @notice Emitted when the StateBridge sets the gas limit for SetRootHistoryExpiry
+    /// @param _lineaGasLimit The new _lineaGasLimit  for SetRootHistoryExpiry
+    event SetGasLimitSetRootHistoryExpiry(uint32 _lineaGasLimit);
+
+    /// @notice Emitted when the StateBridge sets the gas limit for transferOwnershipOp
+    /// @param _lineaGasLimit The new _lineaGasLimit  for transferOwnershipOptimism
+    event SetGasLimitTransferOwnershipLinea(uint32 _lineaGasLimit);
```

Comment:
Comments still mention Optimism chain, I guess it was not intended

Reply:
Sorry, left you a bit of a work. If it's too late, I think we can finish it tomorrow